### PR TITLE
port to Python 3

### LINKED ISSUE #493: Support Python 3.x
The world is changing.  We should change with it.

### PR DESCRIPTION
**Pull request by whit537**
Closes #493.

### CODE REVIEW

**Comment by whit537** [comment]
Builds on #520.

**Comment by whit537** [comment]
Currently hitting an issue with cookie handling: http://bugs.python.org/issue11315.

**Comment by pjz** [comment]
I did a bunch of this work a while ago, but never pushed it up because I got stuck.  It's on the py3 branch.

**Comment by whit537** [comment]
https://github.com/gratipay/aspen.py/tree/py3

File: aspen/testing/client.py [hunk 1]
```diff
@@ -122,7 +119,7 @@ def hit(self, method, path='/', data=None, body=b'', content_type=MULTIPART_CONT
         if content_type is MULTIPART_CONTENT:
             body = encode_multipart(BOUNDARY, data)
 
-        environ = self.build_wsgi_environ(method, path, body, str(content_type), **headers)
+        environ = self.build_wsgi_environ(method, path, body, six.binary_type(content_type), **headers)
```
Comment (Changaco) [inline aspen/testing/client.py]: `six.binary_type` is just `bytes`, and you shouldn't call it.

**Comment by whit537** [comment]
Closing this PR. Porting to Python 3 has proven to be a bridge too far for us. :disappointed:

**Comment by pjz** [comment]
I think I'd rather say that we're going to port to python 3 *after* we do the pending major restructuring.

**Comment by whit537** [comment]
> I think I'd rather say that we're going to port to python 3 *after* we do the pending major restructuring.

Do you think this PR is worth keeping open until then? Seems to me that the restructuring is major enough that it's not worth keeping this PR. We should start a new one.

**Comment by pjz** [comment]
Agreed; sorry, I thought this was the issue, not the PR.

**Pull request closed without merging**